uobtheatre/bookings/schema.py

```diff
@@ -281,7 +332,47 @@ def resolve_mutation(self, root, info, booking_id, price, nonce):
         return PayBooking(booking=booking, payment=payment)
 
 
+class CheckInBooking(AuthRequiredMixin, SafeMutation):
+    performance = graphene.Field("uobtheatre.productions.schema.PerformanceNode")
+    booking = graphene.Field(BookingNode)
+
+    class Arguments:
+        booking_reference = graphene.String(required=True)
+        performance_id = IdInputField(required=True)
+        tickets = graphene.List(CheckInTicketInput, required=True)
+
+    @classmethod
+    def resolve_mutation(self, root, info, booking_reference, tickets, performance_id):
+        performance = Performance.objects.get(id=performance_id)
+        booking = Booking.objects.get(reference=booking_reference)
+
+        # check if the booking pertains to the correct performance
+        if booking.performance != performance:
+            raise GQLFieldException(
+                message="The booking performance does not match the given performance."
+            )
+            # raise booking performance does not match performance given
+        ticket_objects = list(map(lambda ticket: ticket.to_ticket(), tickets))
+        # loop through the ticket IDs given
+        for ticket in ticket_objects:
+            # Check the ticket booking matches the given booking
+            if ticket.booking == booking:
+                ticket.check_in()
+            else:
+                raise GQLFieldException(
+                    message="The ticket booking does not match the mutation booking."
+                )
```

Comment:
To be consistent we should probably do the same style here:
```suggestion
            if ticket.booking != booking:
                raise GQLFieldException(
                    message="The ticket booking does not match the mutation booking."
                )
                
            ticket.check_in()
```